insights/parsers/initscript.py

```diff
@@ -0,0 +1,84 @@
+"""
+InitScript - files ``/etc/rc.d/init.d/*``
+=========================================
+
+InitScript is a parser for the initscripts in ``/etc/rc.d/init.d``.
+
+Because this parser read multiple files, the initscripts are stored as a list
+within the parser and need to be iterated through in order to find specific
+initscripts.
+
+Examples:
+
+    >>> for initscript in shared[InitScript]: # Parser contains list of all initscripts
+    ...     print "Name:", initscript.file_name
+    ...
+    Name: rhnsd
+
+"""
+
+from . import SkipException
+from .. import Parser, parser
+import re
+
+CHKCONFIG_REGEX = re.compile(r"^#\s+chkconfig:\s+.*$")
+LSB_REGEX = re.compile(r"^#\s+Provides:\s+.*$")
+
+SHEBANG_REGEX = re.compile(r"^#!\s*/.*$")
+STARTSTOPSTATUS_REGEX = re.compile(r"\b(start|stop|status)\b")
+
+
+class EmptyFileException(SkipException):
+    pass
+
+
+class NotInitscriptException(SkipException):
+    pass
+
+
+@parser("initscript")
+class InitScript(Parser):
+    """
+    Parse initscript files. Each item is a dictionary with following fields:
+
+    * ``file_name`` - initscript name
+    * ``file_path`` - initscript path (without leading '/')
+    * ``file_content`` - initscript content (list of lines)
```

Comment:
If you wanted, you could convert this to an attributes section in the docstring, for example:
```
Attributes:
    file_name (str): initscript name
    file_path (str): initscript path (without leading '/')
    file_content (list): initscript content (list of lines)
```